### ISSUE TITLE
Android sometime crashed on the isValidZipFile() 

### ISSUE THREAD

**Issue by foobnix** [srikanth-lingala/zip4j#171]
1. case
I can suppose this is a very long time operation,
Is it possible to make method **isValidZipFile()** very fast and low resource required?

at libcore.io.BlockGuardOs.lseek (BlockGuardOs.java:186)
  at java.io.RandomAccessFile.seek (RandomAccessFile.java:615)
  at net.lingala.zip4j.headers.HeaderReader.seekInCurrentPart (HeaderReader.java:739)
  at net.lingala.zip4j.headers.HeaderReader.readEndOfCentralDirectoryRecord (HeaderReader.java:113)
  at net.lingala.zip4j.headers.HeaderReader.readAllHeaders (HeaderReader.java:74)
  at net.lingala.zip4j.ZipFile.readZipInfo (ZipFile.java:964)
  at net.lingala.zip4j.ZipFile.isValidZipFile (ZipFile.java:912)

**Comment by foobnix** [comment]
2) case **isValidZipFile**

 at libcore.io.Linux.readBytes (Native method)
  at libcore.io.Linux.read (Linux.java:184)
  at libcore.io.BlockGuardOs.read (BlockGuardOs.java:254)
  at libcore.io.IoBridge.read (IoBridge.java:501)
  at java.io.RandomAccessFile.readBytes (RandomAccessFile.java:385)
  at java.io.RandomAccessFile.read (RandomAccessFile.java:414)
  at java.io.RandomAccessFile.readFully (RandomAccessFile.java:473)
  at java.io.RandomAccessFile.readFully (RandomAccessFile.java:453)
  at net.lingala.zip4j.util.RawIO.readIntLittleEndian (RawIO.java:80)
  at net.lingala.zip4j.headers.HeaderReader.readEndOfCentralDirectoryRecord (HeaderReader.java:115)
  at net.lingala.zip4j.headers.HeaderReader.readAllHeaders (HeaderReader.java:74)
  at net.lingala.zip4j.ZipFile.readZipInfo (ZipFile.java:964)
  at net.lingala.zip4j.ZipFile.isValidZipFile (ZipFile.java:912)

**Comment by srikanth-lingala** [comment]
before I start to look into this, can you please let me know the version of zip4j that you are using?

**Comment by foobnix** [comment]
2.5.0

**Comment by srikanth-lingala** [comment]
A few questions:

1. In the case that you see those exceptions, was it a valid zip file?
2. Is that the complete stack trace you get? Is there no exception message? From the stack trace you posted, it is not quite clear what the issue was.
3. Are you able to reproduce this issue? If yes, can you check how the behaviour is with version 2.3.1.

My best guess at the moment, is that zip4j was not able to find certain headers at the place it expects it to find. In this case, zip4j tries to read backwards byte-by-byte and see if it found headers anywhere else. And reading in such manner is probably causing the issue you are experiencing. Answers to the above question can help me narrow down the cause of this issue.

**Comment by foobnix** [comment]
> In the case that you see those exceptions, was it a valid zip file?
I can't reproduce this exception, I just receive them from users (as logs in GPlay console)

>Is that the complete stack trace you get? Is there no exception message?
yes this is a complete stack trace

>If yes, can you check how the behavior is with version 2.5.1.
I will prepare a new app build with the final version

My idea is next: *you should make the method **isValidZipFile()** very fast and optimized*

I am using in this way:
```
if (zp.isValidZipFile()) {
     fileHeaders = zp.getFileHeaders();
    ....
}
```

**Comment by srikanth-lingala** [comment]
From your reply above:

> If yes, can you check how the behavior is with version 2.5.1.

I actually wrote in my post to use 2.3.1 and not 2.5.1. But I guess you can't rollback to 2.3.1 because these logs are from your users, and not directly from you. 

I have an idea that will most likely improve the performance of this. I will try to include it in the next release.

**Comment by foobnix** [comment]
> actually wrote in my post to use 2.3.1 

Sorry, I can't do that because You have made critical fixed in later releases
I will wait for new release

**Comment by srikanth-lingala** [comment]
@foobnix I have made a fix for this. I think that should fix the issue. I included the fix in v2.5.2 released today. Let me know if you still see these issues.
